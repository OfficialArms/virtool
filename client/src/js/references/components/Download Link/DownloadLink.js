import styled from "styled-components";
import { BoxGroupSection } from "../../../base";

export const DownloadLink = styled(BoxGroupSection)`
    align-items: center;
    display: flex;

    a {
        margin-left: auto;
    }
`;
