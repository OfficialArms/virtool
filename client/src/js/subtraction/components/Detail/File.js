import React from "react";
import styled from "styled-components";
import { BoxGroupSection } from "../../../base";
import { byteSize } from "../../../utils/utils";
import { fontWeight } from "../../../app/theme";

const StyledBoxGroupSection = styled(BoxGroupSection)`
    align-items: center;
    display: flex;

    a {
        margin-right: auto;
        font-weight: ${fontWeight.thick};
    }
`;

export const SubtractionFile = ({ file }) => {
    return (
        <StyledBoxGroupSection>
            <a href={file.download_url}>{file.name}</a>
            <strong>{byteSize(file.size)}</strong>
        </StyledBoxGroupSection>
    );
};
