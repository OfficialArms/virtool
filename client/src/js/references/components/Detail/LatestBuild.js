import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { NoneFoundSection, RelativeTime } from "../../../base";
import { DownloadLink } from "../Download Link/DownloadLink";

export const LatestBuild = ({ id, latestBuild }) => {
    if (latestBuild) {
        return (
            <DownloadLink>
                <div>
                    <strong>
                        <Link to={`/refs/${id}/indexes/${latestBuild.id}`}>Index {latestBuild.version}</Link>
                    </strong>
                    <span>
                        &nbsp;/ Created <RelativeTime time={latestBuild.created_at} /> by {latestBuild.user.id}
                    </span>
                </div>
                {latestBuild.has_json && (
                    <a href={`/download/indexes/${latestBuild.id}`} download>
                        Download Index
                    </a>
                )}
            </DownloadLink>
        );
    }

    return <NoneFoundSection noun="index builds" />;
};

LatestBuild.propTypes = {
    id: PropTypes.string,
    latestBuild: PropTypes.object
};
