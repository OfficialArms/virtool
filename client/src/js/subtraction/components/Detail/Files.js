import React from "react";
import { get } from "lodash-es";
import { connect } from "react-redux";
import { BoxGroupHeader, BoxGroup, NoneFound } from "../../../base";
import { SubtractionFile } from "./File";

export const SubtractionFiles = ({ files }) => {
    if (files.length == 0) {
        return <NoneFound noun="subtraction files" />;
    }

    return (
        <BoxGroup>
            <BoxGroupHeader>
                <h2>Files</h2>
                <p>Data files available to workflows using this subtraction.</p>
            </BoxGroupHeader>
            {files.map(file => (
                <SubtractionFile file={file} key={file.id} />
            ))}
        </BoxGroup>
    );
};

const mapStateToProps = state => {
    return {
        files: get(state, "subtraction.detail.files", [])
    };
};

export default connect(mapStateToProps)(SubtractionFiles);
