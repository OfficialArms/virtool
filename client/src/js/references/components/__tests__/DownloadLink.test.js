import { DownloadLink } from "../Detail/DownloadLink";
import { screen } from "@testing-library/react";

describe("<DownloadLink />", () => {
    const props = {
        onSelect: jest.fn(),
        id: 1
    };

    it("should render", () => {
        const wrapper = shallow(<DownloadLink {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should download the correct link when clicked", () => {
        renderWithProviders(<DownloadLink {...props} />);
        const link = screen.getByText("Download Index").closest("a");
        expect(link).toHaveAttribute("href", `/download/indexes/${props.id}`);
    });
});
