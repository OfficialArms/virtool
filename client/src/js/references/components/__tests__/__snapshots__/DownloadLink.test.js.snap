// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadLink /> should render 1`] = `
<a
  download={true}
  href="/download/indexes/1"
>
  Download Index
</a>
`;
